query/compute.sp

```diff
@@ -0,0 +1,284 @@
+      select
+      *
+      from
+        terraform_resource
+      where
+        type = 'azurerm_virtual_machine' and (arguments -> 'os_profile_linux_config') is not null
+    ), vm_extensions as (
+        select
+          *
+        from
+          terraform_resource
+        where
+          type = 'azurerm_virtual_machine_extension'
+    ),
+    vm_guest_configuration as (
+      select
+        split_part((b.arguments ->> 'virtual_machine_id'), '.', 2) as vm_name
+      from
+        all_linux_vm as a left join vm_extensions as b on  split_part((b.arguments ->> 'virtual_machine_id'), '.', 2) = a.name
+      where
+        (b.arguments ->> 'publisher') = 'Microsoft.GuestConfiguration'
+    )
+    select
+      type || ' ' || name as resource,
+      case
+        when d.vm_name is null then 'alarm'
+        else 'ok'
+      end as status,
+      name || case
+        when d.vm_name is null then ' have guest configuration extension not installed'
+        else ' have guest configuration extension installed'
+      end || '.' reason
+      ${local.tag_dimensions_sql}
+      ${local.common_dimensions_sql}
+    from
+      all_linux_vm as c left join vm_guest_configuration as d on c.name = d.vm_name;
+  EOQ
+}
+
+query "compute_vm_and_scale_set_encryption_at_host_enabled" {
+  sql = <<-EOQ
+    (select
+      type || ' ' || name as resource,
+      case
+        when (arguments -> 'encryption_at_host_enabled') is null then 'alarm'
+        when (arguments -> 'encryption_at_host_enabled')::boolean then 'ok'
+        else 'alarm'
+      end status,
+      name || case
+        when (arguments -> 'encryption_at_host_enabled') is null then ' ''encryption_at_host_enabled'' is not defined'
+        when (arguments -> 'encryption_at_host_enabled')::boolean then ' encryption at host enabled'
+        else ' encryption at host disabled'
+      end || '.' reason,
+      path || ':' || start_line
+      ${local.tag_dimensions_sql}
+      ${local.common_dimensions_sql}
+    from
+      terraform_resource
+    where
+      type = 'azurerm_linux_virtual_machine_scale_set' or
+      type = 'azurerm_windows_virtual_machine_scale_set'
+    )
+    union
+    (
+    select
+      type || ' ' || name as resource,
+      case
+        when (arguments -> 'encryption_at_host_enabled') is null then 'alarm'
+        when (arguments -> 'encryption_at_host_enabled')::boolean then 'ok'
+        else 'alarm'
+      end status,
+      name || case
+        when (arguments -> 'encryption_at_host_enabled') is null then ' ''encryption_at_host_enabled'' is not defined'
+        when (arguments -> 'encryption_at_host_enabled')::boolean then ' encryption at host enabled'
+        else ' encryption at host disabled'
+      end || '.' reason
+      ${local.tag_dimensions_sql}
+      ${local.common_dimensions_sql}
+    from
+      terraform_resource
+    where
+      type = 'azurerm_windows_virtual_machine' or
+      type = 'azurerm_linux_virtual_machine'
+    )
+  EOQ
+}
```

Comment:
this gives error > `each UNION query must have the same number of columns (SQLSTATE 42601)`